feg/gateway/services/session_proxy/credit_control/gx/gx_client.go

```diff
@@ -252,6 +251,16 @@ func (gxClient *GxClient) createCreditControlMessage(
 		m.NewAVP(avp.FramedIPv6Prefix, avp.Mbit, 0, datatype.OctetString(Ipv6PrefixFromMAC(request.HardwareAddr)))
 	}
 
+	// IPv6
+	if ipv6 := net.ParseIP(request.IPv6Addr); ipv6 != nil {
```

Comment:
Would it be acceptable to have both ipv4 & ipv6 set?

Reply:
per what I can read at `TS 29.212` it mentions in several occasions that it can have v4 or/and v6

```
IP-CAN session: association between a UE and an IP network.
The association is identified by one or more UE Ipv4 addresses/ and/or Ipv6 prefix 
together with a UE identity information, if available, and a PDN represented by a 
PDN ID (e.g. an APN). An IP-CAN session incorporates one or more IP-CAN bearers. 
Support for multiple IP-CAN bearers per IP-CAN session is IP-CAN specific. An IP-CAN
session exists as long as the related UE Ipv4 address and/or Ipv6 prefix are assigned 
and announced to the IP network.
```

```
4.5.1 Request for PCC rules 
1) At IP-CAN session establishment:
The PCEF shall send a CC-Request with CC-Request-Type AVP set to the value 'INITIAL_REQUEST'. The
PCEF shall supply user identification within the Subscription-Id AVP and other attributes to allow
the PCRF to identify the rules to be applied. The other attributes shall include the type of
IP-CAN within the IP-CANType AVP, the type of the radio access technology, if available, 
within the RAT-Type AVP, the PDN information, if available, within the Called-Station-Id AVP, 
the PDN connection identifier, if available,within the PDN-Connection-ID AVP, the UE Ipv4 
address within the Framed-IP-Address and/or the UE Ipv6 prefix within the Framed-Ipv6-Prefix 
AVP and the UE time zone information within 3GPP-MS-TimeZone AVP, if available
```


[https://www.etsi.org/deliver/etsi_ts/129200_129299/129212/12.06.00_60/ts_129212v120600p.pdf](https://www.etsi.org/deliver/etsi_ts/129200_129299/129212/12.06.00_60/ts_129212v120600p.pdf)